auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -1170,4 +1193,8 @@ protected FlowParameters getFlowParams() {
                     mEnableAnonymousUpgrade);
         }
     }
+
+    public class AnonymousAuthProvider {
+        public static final String PROVIDER_ID = "anonymous";
+    }
```

Comment:
Not quite sure I see why we need this... Can't we just use `AnonymousAuthProvider.PROVIDER_ID`?

Reply:
Where is this AnonymousAuthProvider that you speak of? :smile:

Reply:
😆❤️

Reply:
Oh lol, you named it so well I thought you were joking. 😊 Anyway... since Firebase _doesn't_ have a built-in `AnonymousAuthProvider`, shouldn't the [`FirebaseAuthProvider`](https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseAuthProvider) essentially be the same thing? @samtstern Am I mixing it up or is `FirebaseAuthProvider` intended to represent anon auth?

Reply:
No idea, @lsirac might know better than I do!

Reply:
I don't think FirebaseAuthProvider is intended to represent anon auth. Every signed in user has a "firebase" provider.

Reply:
Gotya, then I think we should stick to a field in AuthUI directly. We used to have stuff like this:
```
-    /**
-     * Provider identifier for email and password credentials, for use with {@link
-     * SignInIntentBuilder#setAvailableProviders(List)}.
-     */
-    public static final String EMAIL_PROVIDER = EmailAuthProvider.PROVIDER_ID;
```

So we could do `ANONYMOUS_PROVIDER`. Also, does it need to be publicly accessible? I think we could mark it with `@RestrictTo`.

Reply:
So we'd do `AuthUI.ANONYMOUS_PROVIDER` instead of `AuthUI.AnonymousAuthProvider.PROVIDER_ID`? and yeah we can mark it with `@RestrictTo`

Reply:
Yup! 😊